src/main.cpp

```diff
@@ -473,6 +477,31 @@ static void _assert_permissions(const string &sota_config_dir)
 	unlink(test_file.c_str());
 }
 
+static void _assert_not_registered(const string &sota_config_dir)
+{
+	string path = sota_config_dir + "/sql.db";
+	if (access(path.c_str(), F_OK ) == 0 ) {
+		cerr << "ERROR: Device appears to already be registered in " << path << endl;
+		exit(EXIT_FAILURE);
+	}
+}
+
+static void _assert_not_running() {
+	string out = _spawn("ps -ef");
+	size_t pos = 0;
+	while ((pos = out.find("\n")) != std::string::npos) {
+	    string line = out.substr(0, pos);
+	    if (line.find("aktualizr-lite") != std::string::npos) {
+	    	if (line.find("daemon") != std::string::npos) {
+	    		cerr << "ERROR: aktualizr-lite daemon appears to be running:" << endl;
+			cerr << line << endl;
+			exit(EXIT_FAILURE);
+		}
+	    }
+	    out.erase(0, pos + 1);
```

Comment:
Is there a particular reason to search in lines for `aktualizr-lite` instead of searching just in the overall output?

Reply:
2 reasons:
1) I wanted to make sure the line also has "daemon" in it.
2) I wanted to be able to try and share the actual process we are finding so a user will know what they need to kill